_posts/2023-06-23-run-and-debug-rust-verification-harnesses-natively-on-vscode.md

```diff
@@ -0,0 +1,178 @@
+
+## A Simple Example
+
+To show some of the features in the extension, we will work through a familiar example, the rectangle example that we introduced in our [first blog post](https://model-checking.github.io/kani-verifier-blog/2022/05/04/announcing-the-kani-rust-verifier-project.html).
+
+```rust
+#[derive(Debug, Copy, Clone)]
+struct Rectangle {
+    width: u64,
+    height: u64,
+}
+
+impl Rectangle {
+    fn can_hold(&self, other: &Rectangle) -> bool {
+        self.width > other.width && self.height > other.height
+    }
+
+    fn stretch(&self, factor: u64) -> Option<Self> {
+        let w = self.width.checked_mul(factor)?;
+        let h = self.height.checked_mul(factor)?;
+        Some(Rectangle { width: w, height: h })
+    }
+}
+```
+
+In order to verify properties about the rectangle, we wrote a verification harness.
+The harness tried to verify that when the rectangle is stretched, it can hold another rectangle of its original size (dimensions).
+If verified succesfully, this means that for any given stretch factor and any height and width of the original rectangle, this property holds true.
+
+```rust
+#[kani::proof]
+pub fn stretched_rectangle_can_hold_original() {
+    let original = Rectangle { width: kani::any(), height: kani::any() };
+    let factor = kani::any();
+    if let Some(larger) = original.stretch(factor) {
+        assert!(larger.can_hold(&original));
+    }
+}
+```
+
+#### How Kani is used currently
+
+The current way of interacting with Kani is through the command line.
+Users invoke `cargo kani` and specify the harness they want to verify.
+Kani produces text-based output that tells you whether your verification attempt has succeeded or failed.
+
+```
+$ cargo kani --harness stretched_rectangle_can_hold_original
+# --snip--
+Check 2: verification::stretched_rectangle_can_hold_original.assertion.1
+         - Status: FAILURE
+         - Description: "assertion failed: larger.can_hold(&original)"
+         - Location: src/main.rs:36:13 in function verification::stretched_rectangle_can_hold_original
+# --snip--
+VERIFICATION:- FAILED
+```
+
+## Introducing the Kani VS Code Extension
+
+The [Kani VS Code extension](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension) offers a hassle-free and seamless integration into Visual Studio Code, making it more convenient to write and debug harnesses.
+As you write Kani harnesses, the extension detects them and conveniently displays them within your testing panel.
+The extension offers detailed diagnostics, feedback about verification failures, error messages, and stack traces.
+This empowers our users to find bugs and verify their code quicker.
+You can install the extension from the [webpage](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension) on VS Code marketplace or by searching for `Kani` in the extensions tab of your VS Code window.
```

Comment:
I'd move this whole paragraph up to the the top of this post and combine it with the introduction before the example. Once you've introduced the example you don't want to have any more such general statements you just want to stick with the "tutorial" portion, it's a better flow.